### PR TITLE
Add -snapshot option for snapshotting table state

### PR DESCRIPTION
**Pull request by bruth**
The purpose of this is to support initializing the state of a stream by
snapshotting the initial state of the table and as well as creating
snapshots of the table on regular intervals. This reduces the overhead
for consumers to produce the current state by reading the stream. A
snapshot enables a consumer to start at the last known snapshot and read
from there rather than the beginning of the stream.

Signed-off-by: Byron Ruth <b@devel.io>

### CODE REVIEW

**Comment by bruth** [comment]
This also adds a `time` value to events (sort of important for an event!). Here is a sample usage:

```bash
% diff-table -csv1 example/file1.csv -key id -snapshot | jq .
{
  "type": "row-stored",
  "time": 1520119177,
  "offset": 1,
  "key": {
    "id": "1"
  },
  "data": {
    "color": "Blue",
    "gender": "Male",
    "id": "1",
    "name": "John"
  }
}
{
  "type": "row-stored",
  "time": 1520119177,
  "offset": 2,
  "key": {
    "id": "2"
  },
  "data": {
    "color": "Red",
    "gender": "Female",
    "id": "2",
    "name": "Pam"
  }
}
{
  "type": "row-stored",
  "time": 1520119177,
  "offset": 3,
  "key": {
    "id": "3"
  },
  "data": {
    "color": "Yellow",
    "gender": "Female",
    "id": "3",
    "name": "Sam"
  }
}
```

**Comment by bruth** [comment]
@aaron0browne when you have a few minutes can you review this.

File: README.md [hunk 1]
```diff
@@ -115,7 +117,43 @@ diff-table \
...
+
+  // Offset of the row in the batch, if applicable. This can also be used as an
+  // ordering mechanism.
+  "offset": 1,
+
+  // Key of the row this event pertains to, if applicable.
+  "key": {...},
+
+  // Full snapshot of the row. This will be present row row-added events, but
+  // can be optionally included for row-changed and row-removed events.
```
Comment (gracecodes144) [inline README.md]: > can be optionally included for row-changed and row-removed events

Is this option already implemented or did I miss it in these changes?

Comment (bruth) [reply README.md]: Good catch! Adding..

Comment (bruth) [reply README.md]: Added a new option to support this.

**Review by gracecodes144 — APPROVED** [review]

**Pull request merged**